### DISCUSSION TITLE
Adding child routes to Index routes

### DISCUSSION THREAD

**Discussion by elmeerr** [remix-run/remix#5512] [Q&A] [answered]
The documentation says:  `index routes cannot have child routes` and today I got a scenario related to this statement.

I have the following:
```
routes/
 users/
    index.tsx
```

and I would like to have a modal opened, without state, so I would open by going to a route `/users/new`. I would expect to see the content of `users/index.ts` behind the modal, but as the documentation says, this is not possible.
I could achieve what I want by creating a `home` folder and do the nesting there by using [this approach](https://github.com/remix-run/remix/discussions/3464) but I really didn't want to create another path to my url. Because of that I try somethings and I ended up doing the following:

```
users/
 index.tsx
 __home.tsx
 __home
     new.tsx
```

inside of `index.tsx` I have my root component:
<img width="321" alt="Screenshot 2023-02-18 at 12 14 36" src="https://user-images.githubusercontent.com/4071037/219857284-118b80e2-35ee-4581-afb1-5ab4f0f3bfe0.png">

and inside of `__home.tsx` my root component + `Outlet`
<img width="319" alt="Screenshot 2023-02-18 at 12 16 13" src="https://user-images.githubusercontent.com/4071037/219857324-4f97d007-f345-4dd0-b7df-34a83a839f5f.png">

with this setup I can access: `/users` (`index.tsx)` and `users/new` (`__home/new.tsx`) and have the same result as I would have by nesting routes inside of a `home` folder, without having the extra path on my URL, thanks to pathless layouts 😅 

is this a valid approach? or should I just "follow the rules" and use a `home` folder instead? Any thoughts would be appreciated.

Regards,

**Comment by sergiodxa** [comment, marked as answer]
You could use a search param `?new` and then check if the search param is there open the modal.

**Comment by elmeerr** [comment]
@sergiodxa but then I would need to use state on my component, right? I am trying to avoid state and use nested routes in order to work with modals...not sure if it its the best / maintainable approach but I kind of like using it on Remix.

**Reply by sergiodxa** [reply]
The search param is your state, no need to add a useState, you can read the search params in the loader or directly in the component.

**Reply by elmeerr** [reply]
by state I mean having an `if` clause to control the state of my dialog...I was mostly trying to have input about the approach I took...is kind of hacky but it gives me the result I expect. Your suggestion is also good I will take into consideration because I am trying to follow a pattern when it comes to dialogs...my options are:
- "follow the rules" and add a `home` folder
- use your suggestion
- keep my approach

thanks for taking the time to answer.
